ek9000App/src/devEK9000.cpp

```diff
@@ -69,6 +69,17 @@ std::list<devEK9000*>& GlobalDeviceList() {
 bool devEK9000::debugEnabled = false;
 int devEK9000::pollDelay = 200;
 
+// This is a big hack for safety reasons. This will force you to use the define PDO macro for every terminal type at
+// least once, so we can catch size mismatches easily. If LTO removes this symbol, it's not the end of the world, it'll
+// still serve its purpose in non-optimized debug builds
+void __PDOHack() {
+	__PDO_CheckHack();
```

Comment:
It's an interesting way of doing things. Perhaps it's a hack, but it's one that appears to compile, work portably (well, CI passed), and guard against coding mistakes.

Admittedly, your C++-fu has long-since eclipsed my own, so I have no good alternative suggestions on how to better structure things without ugly macros and something like `__PDOHack` above.

Reply:
> Admittedly, your C++-fu has long-since eclipsed my own

I think you can call it C++-itis :grin:. I've decided to templatize this a bit, which should reduce code size a wee bit. Also rewriting the generate.py script at this point because it's so ugly haha.